src/caselawclient/client_helpers/__init__.py

```diff
@@ -0,0 +1,78 @@
+import json
+from enum import Enum
+from typing import Optional, TypedDict
+
+from typing_extensions import NotRequired
+
+
+class AnnotationDataDict(TypedDict):
+    type: str
+    calling_function: str
+    message: NotRequired[str]
+
+
+class VersionType(Enum):
+    """Valid types of version."""
+
+    SUBMISSION = "submission"
+    """ This version has been created as a result of a submission of a new document. """
+
+    ENRICHMENT = "enrichment"
+    """ This version has been created through an enrichment process. """
+
+    EDIT = "edit"
+    """ This version has been created as the result of a manual edit. """
+
+
+class VersionAnnotation:
+    """A class holding structured data about the reason for a version."""
+
+    def __init__(self, version_type: VersionType, message: Optional[str] = None):
+        """
+        :param version_type: The type of version being created
+        :param message: A human-readable string containing information about the version which can't be expressed in the
+            structured data.
+        """
+        self.version_type = version_type
+        self.message = message
```

Comment:
oops, tiny thing Nick which i've just noticed when trying to use this in anger - we should probably set self.calling_function = None here or we end up with an AttributeError later when trying to use the annotation. In practice, this shouldn't happen, but it leads to some complications in tests